gateway/src/shard/processor/emit.rs

```diff
@@ -18,13 +19,12 @@ pub async fn bytes(listeners: Listeners<Event>, bytes: &[u8]) {
 }
 
 pub fn event(listeners: Listeners<Event>, event: Event) {
-    let mut executor = DefaultExecutor::current();
+    // TODO: fix this so it spawns on the current executor.
+    // let mut executor = DefaultExecutor::current();
```

Comment:
Why can't this spawn on the current executor?

Reply:
Just asked some tokio people about it and it looks like I have done the correct thing, I will remove the comments tomorrow.